README.md

```diff
@@ -3,25 +3,93 @@
+  * [webmozart/assert](https://github.com/webmozarts/assert)
+* Otherwise: Run `composer update` inside the `de.systopia.remotetools` folder.
+* Enable the extension. (Either via UI or `cv ext:enable de.systopia.remotetools`)
+
+For the frontend part you can use the [civiremote Drupal module](https://github.com/systopia/civiremote).
 
 ## Usage
 
-The following modules rely on this extension:
+You can use this extension via other extensions that rely on Remote Tools:
 
-* [Event Messages Extension](https://github.com/systopia/de.systopia.eventmessages)
 * [Remote Events Extension](https://github.com/systopia/de.systopia.remoteevent)
-* (more to follow) 
+* [Remote Activity Extension](https://github.com/systopia/de.systopia.remoteactivity)
 
-## Requirements
+### Providing own remote entities
+
+If you want to provide an own remote entity that accesses the entity `MyEntity`,
+you can do the following:
+
+* Create the class `\Civi\Api4\RemoteMyEntity` (name can be freely chosen) that extends [`Civi\RemoteTools\Api4\AbstractRemoteEntity`](Civi/RemoteTools/Api4/AbstractRemoteEntity.php).
+  * Implement the method `permissions()` to restrict access to your remote API user. (See [CiviCRM documentation](https://docs.civicrm.org/dev/en/latest/security/permissions/#apiv4).)
+* Create a class that extends [`\Civi\RemoteTools\EntityProfile\AbstractRemoteEntityProfile`](Civi/RemoteTools/EntityProfile/AbstractRemoteEntityProfile.php). (For read only entities use [`\Civi\RemoteTools\EntityProfile\AbstractReadOnlyRemoteEntityProfile`](Civi/RemoteTools/EntityProfile/AbstractReadOnlyRemoteEntityProfile.php) instead.) The various methods are documented in [`\Civi\RemoteTools\EntityProfile\RemoteEntityProfileInterface`](Civi/RemoteTools/EntityProfile/RemoteEntityProfileInterface.php).
+  * Add the public constants `NAME` (set to e.g. `my-profile`, `default`, or also an empty string), `ENTITY_NAME` (set to `MyEntity`), `REMOTE_ENTITY_NAME` (set to `RemoteMyEntity`).
+  * Use the trait [`\Civi\RemoteTools\EntityProfile\Traits\ConstProfileMetadataTrait`](Civi/RemoteTools/EntityProfile/Traits/ConstProfileMetadataTrait.php).
+* Register the class as service in the service container via [`hook_civicrm_container`](https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_container/) like this:
+```php
+$container->autowire(MyRemoteEntityProfile::class)
+  ->addTag(MyRemoteEntityProfile::SERVICE_TAG);
+```
+
+Now you should be able to access data via APIv4, e.g.
+
+```php
+civicrm_api4('RemoteMyEntity', 'get', ['profile' => 'my-profile']);
+```
+
+(Use the APIv4 explorer to test.)
+
+Most methods in the profile class have a parameter named `$contactId`. This
+contains the remote contact ID resolved to a CiviCRM contact ID or `NULL` if no
+remote contact ID was given. If a given remote contact ID could not be resolved,
+access is denied. The resolving is done using the extension
+[de.systopia.identitytracker](https://github.com/systopia/de.systopia.identitytracker).
```

Comment:
Is this suggesting that there's no support for anonymous API calls to any remote entity based on that structure?

Reply:
Anonymous calls are possible:
> [...] resolved to [...] `NULL` if no remote contact ID was given.

Access is denied if a **given** remote contact ID could not be resolved.

Reply:
Thanks for clarifying, my bad!